### PR TITLE
fix: CXMC-23462 Page display incorrectly when token invalid

### PR DESCRIPTION
**Pull request by i53577**
a property is needed in backend. 
```
corsfilter.commercewebservices.exposedHeaders=x-anonymous-consents occ-personalization-id occ-personalization-time www-authenticate
```

this property not locate in any extension in MCS, but a sample property in cxmc-product-pipeline.
https://github.tools.sap/cx-commerce/cxmc-product-pipeline/blob/develop/galectin/resources/tasks/local.properties.tmpl#L37

### CODE REVIEW

File: projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts [hunk 1]
```diff
@@ -65,7 +65,8 @@ export class AuthInterceptor implements HttpInterceptor {
                 } else if (
                   // Refresh the expired token
                   // Check if the OAuth endpoint was called and the error is because the refresh token expired
-                  this.errorIsInvalidToken(errResponse)
+                  this.errorIsInvalidToken(errResponse) ||
+                  this.isOidcInvlidToken(errResponse)
```
Comment (scarai-sap) [inline projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts]: maybe insert logic into errorIsInvalidToken?

File: projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.spec.ts [hunk 2]
```diff
@@ -40,6 +41,9 @@ class MockAuthConfigService implements Partial<AuthConfigService> {
   getTokenEndpoint() {
     return '/authorizationserver/token';
   }
+  public getOAuthLibConfig(): AuthLibConfig {
```
Comment (scarai-sap) [inline projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.spec.ts]: looks the new code is not tested by UT?

File: integration-libs/cdc/root/service/cdc-auth-http-header.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  providedIn: 'root',
+})
+export class CdcAuthHttpHeaderService extends AuthHttpHeaderService {
+  protected cdcAuthLogOutService = inject(CdcAuthLogOutService);
+  /**
+   * override handleExpiredRefreshToken() method
+   */
+  public handleExpiredRefreshToken(): void {
+    this.authRedirectService.saveCurrentNavigationUrl();
+    this.cdcAuthLogOutService.cdcLogout().finally(() => {
```
Comment (scarai-sap) [inline integration-libs/cdc/root/service/cdc-auth-http-header.service.ts]: why we need call logout?

**Review by scarai-sap — COMMENTED** [review]
question raised

File: integration-libs/cdc/root/service/cdc-auth-http-header.service.ts (hunk 3 above)
Comment (i53577) [reply integration-libs/cdc/root/service/cdc-auth-http-header.service.ts]: in AuthHttpHeaderService， here calls `this.authService.coreLogout()`, this method won't logout cdc customer. So we need to replace it with `this.cdcAuthLogOutService.cdcLogout()`

File: projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.spec.ts (hunk 2 above)
Comment (i53577) [reply projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.spec.ts]: UT added

File: projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts (hunk 1 above)
Comment (i53577) [reply projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts]: if someday we decide to keep spartacus for a single branch, then seems split it to a function is more easy to do adaptation :)

Comment (scarai-sap) [reply projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts]: as you already have judgement as below so i guess it's safe to put them into one routine? BTW. Does it only for OIDC(I just wonder whether it's best name for the logic)? 

 if (!this.authConfigService.getOAuthLibConfig().disablePKCE && authHeader)

File: integration-libs/cdc/root/guards/cdc-auth-logout.service.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { CdcLogoutGuard } from './cdc-logout.guard';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CdcAuthLogOutService extends CdcLogoutGuard {
```
Comment (vinchen666) [inline integration-libs/cdc/root/guards/cdc-auth-logout.service.ts]: a service should not be a child class of a guard, guard should relies on service. let's follow **composition over Inheritance** principle

Comment (i53577) [reply integration-libs/cdc/root/guards/cdc-auth-logout.service.ts]: Adapted code to let this service not extends a guard. Please have a check.

File: projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts (hunk 1 above)
Comment (i53577) [reply projects/core/src/auth/user-auth/http-interceptors/auth.interceptor.ts]: moved logic to errorIsInvalidToken

**Review by scarai-sap — APPROVED** [review]
LGTM

**Review by vinchen666 — APPROVED** [review]

**Pull request merged**